src/app/api/setAchievement/route.ts

```diff
@@ -25,10 +24,22 @@ export async function POST(req: NextRequest) {
       });
     }
 
+    const foundAchievement = await db
+      .select()
+      .from(achievements)
+      .where(eq(achievements.userId, userId));
```

Comment:
this makes no sense. it just checks if the user has any achievement in the db and then rejects

Reply:
you need to insert ´´´and(eq(achievements.userId, userId), eq(achievements.achievementEnum, achievementEnum))´´´ inside where()